src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -575,15 +575,29 @@ def process_metric_update_namespace(namespace):
 
 
 def datetime_string_type(string):
-    ''' Validates UTC datettime in format '%Y-%m-%d\'T\'%H:%M\'Z\''. '''
-    date_format = '%Y-%m-%dT%H:%MZ'
-    return datetime.strptime(string, date_format).strftime(date_format)
+    """ Validates UTC datettime in accepted format. Examples: 2017-12-31T01:11:59Z,
+    2017-12-31T01:11Z or 2017-12-31T01Z or 2017-12-31 """
+    accepted_date_formats = ['%Y-%m-%dT%H:%M:%SZ', '%Y-%m-%dT%H:%MZ',
+                             '%Y-%m-%dT%HZ', '%Y-%m-%d']
+    for form in accepted_date_formats:
+        try:
+            return datetime.strptime(string, form).strftime(form)
+        except ValueError:  # checks next format
+            pass
+    raise ValueError("Not Valid Date Format")
 
 
 def datetime_type(string):
-    ''' Validates UTC datettime in format '%Y-%m-%d\'T\'%H:%M\'Z\''. '''
-    date_format = '%Y-%m-%dT%H:%MZ'
-    return datetime.strptime(string, date_format)
+    """ Validates UTC datettime in accepted format. Examples: 2017-12-31T01:11:59Z,
+    2017-12-31T01:11Z or 2017-12-31T01Z or 2017-12-31 """
+    accepted_date_formats = ['%Y-%m-%dT%H:%M:%SZ', '%Y-%m-%dT%H:%MZ',
+                             '%Y-%m-%dT%HZ', '%Y-%m-%d']
+    for form in accepted_date_formats:
+        try:
+            return datetime.strptime(string, form)
+        except ValueError:  # checks next format
+            pass
```

Comment:
Using `continue` here instead of `pass` would make the logic more clear and remove the need for the comment. Ultimately they do the same thing in this case.